pkg/server/plugin/datastore/sql/sql.go

```diff
@@ -524,10 +559,10 @@ func deleteBundle(tx *gorm.DB, req *datastore.DeleteBundleRequest) (*datastore.D
 
 	// Get a count of associated registration entries
 	entriesAssociation := tx.Model(model).Association("FederatedEntries")
-	entriesCount := entriesAssociation.Count()
 	if err := entriesAssociation.Error; err != nil {
 		return nil, sqlError.Wrap(err)
 	}
+	entriesCount := entriesAssociation.Count()
```

Comment:
The order of statements is important here. Count() is the method that actually executes the query. As such, the error check must come after.